daemon/daemon_test.go

```diff
@@ -38,3 +40,76 @@ func TestFailedReload(t *testing.T) {
 		t.Fatalf("Should have errored on changed listen port")
 	}
 }
+
+var localServerPort = ":8081"
+var localServerHost = "http://localhost" + localServerPort
+var localProxyHost = "http://localhost:6634"
+
+func setUpDaemonWithLocalServer() error {
+	// Set up a local server that 404s everywhere except one route.
+	mux := http.NewServeMux()
+	mux.HandleFunc("/healthyroute", func(rw http.ResponseWriter, req *http.Request) {
+		rw.Write([]byte("healthy"))
+	})
+	mux.HandleFunc("/", func(rw http.ResponseWriter, req *http.Request) {
+		rw.WriteHeader(http.StatusNotFound)
+		rw.Write([]byte("404"))
+	})
+	go http.ListenAndServe(localServerPort, mux)
+
+	// Set up the daemon to proxy to the local server.
+	conf, err := config.New("../example.yaml")
+	if err != nil {
+		return err
+	}
+	conf.Proxy.Host = localServerHost
+
+	daemon, err := New(conf)
+	if err != nil {
+		return err
+	}
+
+	go daemon.Start()
+	return nil
+}
+
+// testProxyRequest calls the proxy server at the given path and verifies that
+// the request returns the given HTTP status and body content.
+func testProxyRequest(t *testing.T, path string, expectedStatus int, expectedBody string) {
+	resp, err := http.Get(localProxyHost + path)
+	if err != nil {
+		t.Fatalf("Proxy request failed: %s", err.Error())
+	}
+
+	if resp.StatusCode != expectedStatus {
+		t.Fatalf("Response status with path %s does not match expected value.  Actual: %d.  Expected: %d.",
+			path, resp.StatusCode, expectedStatus)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("Couldn't read response body: %s", err.Error())
+	}
+
+	bodyStr := string(body)
+	if bodyStr != expectedBody {
+		t.Fatalf("Response body does not match expected value. Actual: \"%s\" Expected: \"%s\"",
+			bodyStr, expectedBody)
+	}
+}
+
+func TestHealthCheck(t *testing.T) {
+	err := setUpDaemonWithLocalServer()
+	if err != nil {
+		t.Fatalf("DAEMON_SETUP_FAILED: %s", err.Error())
```

Comment:
Is there precedent for the `ALL_CAPS_SNAKE_CASE` error names?  If not (and it seems not, looking through the repo), I'd probably just go with "Test daemon setup failed: %s" or something.